### PR TITLE
Move address caps verification to syntactic

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: sdk/src/types/block/output/account.rs [hunk 1]
```diff
@@ -493,6 +502,8 @@ impl Packable for AccountOutput {
         let features = Features::unpack::<_, VERIFY>(unpacker, &())?;
 
         if VERIFY {
+            verify_restricted_addresses(&unlock_conditions, AccountOutput::KIND, features.native_token(), mana)
```
Comment (Thoralf-M) [inline sdk/src/types/block/output/account.rs]:
```suggestion
            verify_restricted_addresses(&unlock_conditions, Self::KIND, features.native_token(), mana)
```

Comment (Thoralf-M) [reply sdk/src/types/block/output/account.rs]: Same for the others

File: sdk/src/types/block/output/unlock_condition/mod.rs [hunk 2]
```diff
@@ -355,6 +374,53 @@ pub(crate) fn verify_allowed_unlock_conditions(
...
+    unlock_conditions: &UnlockConditions,
+    output_kind: u8,
+    native_token: Option<&NativeTokenFeature>,
+    mana: u64,
+) -> Result<(), Error> {
+    let addresses = unlock_conditions.restricted_addresses();
+
+    for address in addresses {
+        if native_token.is_some() && !address.has_capability(AddressCapabilityFlag::OutputsWithNativeTokens) {
+            return Err(Error::RestrictedAddressCapability);
```
Comment (Thoralf-M) [inline sdk/src/types/block/output/unlock_condition/mod.rs]: Should we return the specific capability that is restricted? Looks annoying to debug without it

Comment (thibault-martinez) [reply sdk/src/types/block/output/unlock_condition/mod.rs]: Sure, done

File: sdk/src/types/block/output/unlock_condition/mod.rs [hunk 3]
```diff
@@ -314,6 +317,22 @@ impl UnlockConditions {
 
         Ok(address)
     }
+
+    /// Returns an iterator over all addresses.
```
Comment (Thoralf-M) [inline sdk/src/types/block/output/unlock_condition/mod.rs]:
```suggestion
    /// Returns an iterator over all addresses except StorageDepositReturn address.
```

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**